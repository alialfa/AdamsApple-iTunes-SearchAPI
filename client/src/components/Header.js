/**
 * Ali_Mongi-CAPSTONE_SEARCH_API-L2T21
 * @Header.js - component that propels a Menubar area
 */
import React from "react";
import Menubar from "./Menubar";

function Header(props) {
  return (
    <div>
      <Menubar companyName="ADAMS" />
    </div>
  );
}

export default Header;
