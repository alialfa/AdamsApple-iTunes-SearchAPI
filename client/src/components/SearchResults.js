/**
 * Ali_Mongi-CAPSTONE_SEARCH_API-L2T21
 * @SearchResults.js - key component that processes search media from App.js and
 *                     creates labels and values,as well as store searchin history
 */
import React, { Component } from "react";
import { Container, Row, Col, Alert } from "react-bootstrap";
import { Link } from "react-router-dom";
import ResultCard from "./ResultCard";

let item1Label = "";
let item2Label = "";
let item3Label = "";
let item4Label = "";
let item5Label = "";
let item6Label = "";
let item7Label = "Genre: ";
let showItem1 = "none";
let showItem2 = "none";
let showItem3 = "none";
let showItem4 = "none";
let showItem5 = "none";
let showItem6 = "none";
let showItem7 = "flex";
let showItem8 = "flex";
let trackItem1,
  collectionItem2,
  ratingItem3,
  softwareItem4,
  genresItem5,
  priceItem6,
  primaryGenreItem7,
  collectionPriceItem8 = "";
let pushCounter = -1;

class SearchResults extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeResults: [],
      resultSuccess: "",
      resultFail: "",
      pushCount: -1,
    };
  }

  componentWillMount() {
    const { searchTerm, kindSelected, media, message } = this.props;

    if (message === "failure") {
      this.setState({
        resultFail: (
          <h1 class="resultFail">
            search_for: [${searchTerm}] ||| <hr></hr>status: [sorry, we couldn't
            find any results for this query...]
          </h1>
        ),
      });
    }
    if (message === "successful") {
      this.setState({
        resultSuccess: `search_for: [${searchTerm}] ||| status: [delivered!]`,
      });

      for (var index = 0; index < media.length; index++) {
        const {
          kind,
          artistName,
          artworkUrl100,
          collectionName,
          trackName,
          collectionPrice,
          primaryGenreName,
          contentAdvisoryRating,
          minimumOsVersion,
          genres,
          price,
        } = media[index];

        let artworkUrl = artworkUrl100.replace(/100x100/g, "1000x1000");

        this.assignResultLabels(kind);
        this.populateLabelsAndValues(
          collectionName,
          trackName,
          collectionPrice,
          primaryGenreName,
          contentAdvisoryRating,
          minimumOsVersion,
          genres,
          price
        );

        let audioKind = kind;
        if (kindSelected === "audiobook") {
          audioKind = media[index].wrapperType;
        }
        if (
          kindSelected === kind ||
          kindSelected === audioKind ||
          kindSelected === "all"
        ) {
          pushCounter++;

          const card = (
            <ResultCard
              artistName={artistName}
              artworkUrl={artworkUrl}
              item1={trackItem1}
              item2={collectionItem2}
              item3={ratingItem3}
              item4={softwareItem4}
              item5={genresItem5}
              item6={priceItem6}
              item7={primaryGenreItem7}
              item8={collectionPriceItem8}
              showItem1={showItem1}
              showItem2={showItem2}
              showItem3={showItem3}
              showItem4={showItem4}
              showItem5={showItem5}
              showItem6={showItem6}
              showItem7={showItem7}
              showItem8={showItem8}
              pushCount={pushCounter}
              handleAddFavorite={this.props.handleAddFavorite}
              resultsHistory={this.props.resultsHistory}
            />
          );

          const favoriteBtn = (
            <i
              as={Link}
              to="/search"
              class="fas fa-star fa-1x resultsCardStarIcon"
              id={pushCounter}
              onClick={this.props.handleAddFavorite}
            />
          );

          const finalCard = (
            <Col xs={3}>
              {card}
              {favoriteBtn}
            </Col>
          );

          this.state.activeResults.push(finalCard);
          this.props.resultsHistory.push(card);
        }
      }
    }

    this.clearDefaults();
  }

  assignResultLabels(kind) {
    switch (kind) {
      case "song":
        item1Label = "Track: ";
        showItem1 = "flex";
        item2Label = "Album: ";
        showItem2 = "flex";
        break;
      case "music-video":
        item1Label = "Video: ";
        showItem1 = "flex";
        break;
      case "feature-movie":
        item1Label = "Movie: ";
        showItem1 = "flex";
        item2Label = "Movie Family: ";
        showItem2 = "flex";
        item3Label = "Rating: ";
        showItem3 = "flex";
        break;
      case "tv-episode":
        item1Label = "TV Show: ";
        showItem1 = "flex";
        item2Label = "Season: ";
        showItem2 = "flex";
        item3Label = "Rating: ";
        showItem3 = "flex";
        break;
      case "podcast":
        item1Label = "Podcast: ";
        showItem1 = "flex";
        break;
      case "software":
        item1Label = "Software Title: ";
        showItem1 = "flex";
        item4Label = "OS Version (minimum): ";
        showItem4 = "flex";
        item6Label = "S/W Price: ";
        showItem6 = "flex";
        showItem7 = "none";
        showItem8 = "none";
        break;
      case "ebook":
        item1Label = "eBook Name: ";
        showItem1 = "flex";
        item5Label = "eBook Genre: ";
        showItem5 = "flex";
        item6Label = "eBook Price: ";
        showItem6 = "flex";
        showItem7 = "none";
        showItem8 = "none";
        break;
      default:
    }
  }

  populateLabelsAndValues(
    collectionName,
    trackName,
    collectionPrice,
    primaryGenreName,
    contentAdvisoryRating,
    minimumOsVersion,
    genres,
    price
  ) {
    trackItem1 = (
      <div className="resultsCardItem">
        <h6>{item1Label}</h6>
        <h5>{trackName}</h5>
      </div>
    );
    collectionItem2 = (
      <div className="resultsCardItem">
        <h6>{item2Label}</h6>
        <h5>{collectionName}</h5>
      </div>
    );
    ratingItem3 = (
      <div className="resultsCardItem">
        <h6>{item3Label}</h6>
        <h5>{contentAdvisoryRating}</h5>
      </div>
    );
    softwareItem4 = (
      <div className="resultsCardItem">
        <h6>{item4Label}</h6>
        <h5>{minimumOsVersion}</h5>
      </div>
    );
    genresItem5 = (
      <div className="resultsCardItem">
        <h6>{item5Label}</h6>
        <h5>{genres}</h5>
      </div>
    );
    priceItem6 = (
      <div className="resultsCardItem" style={{ color: "darkred" }}>
        <h6>{item6Label}</h6>
        <h5>${price}</h5>
      </div>
    );
    primaryGenreItem7 = (
      <div className="resultsCardItem">
        <h6>{item7Label}</h6>
        <h5>{primaryGenreName}</h5>
      </div>
    );
    collectionPriceItem8 = (
      <div className="resultsCardItem" style={{ color: "darkred" }}>
        ${collectionPrice}
      </div>
    );
  }

  clearDefaults() {
    item1Label = "";
    item2Label = "";
    item3Label = "";
    item4Label = "";
    item5Label = "";
    item6Label = "";
    item7Label = "Genre: ";
    showItem1 = "none";
    showItem2 = "none";
    showItem3 = "none";
    showItem4 = "none";
    showItem5 = "none";
    showItem6 = "none";
    showItem7 = "flex";
    showItem8 = "flex";
  }

  render() {
    const { resultSuccess, resultFail, activeResults } = this.state;
    return (
      <Col xs={9}>
        <Container>
          <Alert variant="dark" className="resultAlert">
            {resultSuccess}
          </Alert>
          {resultFail}
          <Row className="searchResults"> {activeResults}</Row>
        </Container>
      </Col>
    );
  }
}

export default SearchResults;
