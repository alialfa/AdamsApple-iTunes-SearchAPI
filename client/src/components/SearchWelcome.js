/**
 * Ali_Mongi-CAPSTONE_SEARCH_API-L2T21
 * @SearchWelcome.js - component that welcomes a user with search instructions
 */
import React from "react";
import { Col, Alert } from "react-bootstrap";

const SearchWelcome = (props) => {
  return (
    <Col xs={9}>
      <div className="searchWelcome">
        <Alert variant="dark">
          <h1>[ New Search Awaits ]</h1>
          <ul>
            <li>Use the side-bar on your left to search.</li>
            <li>Filter your search to a specific category.</li>
            <li>Favorite items by clicking on the star icons.</li>
            <li>Access favorites from the big star on the sidebar.</li>
            <li>
              Thanks for visiting <i>Apple Adam's API </i> :)
            </li>
          </ul>
        </Alert>
      </div>
    </Col>
  );
};

export default SearchWelcome;
