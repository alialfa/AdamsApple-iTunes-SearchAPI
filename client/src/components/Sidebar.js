/**
 * Ali_Mongi-CAPSTONE_SEARCH_API-L2T21
 * @Sidebar.js - component that generates side navigation links and contains a child component for data and cmd inputs
 */
import React from "react";
import { Container, Nav, Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import SearchInput from "./SearchInput";

const Sidebar = (props) => {
  return (
    <Col xs={3}>
      <Nav
        defaultActiveKey="/home"
        className="flex-column col-md-12 d-none d-md-block"
      >
        <Container fluid>
          <div class="sidebar">
            <Nav.Link as={Link} to="/search">
              <i class="fas fa-2x fa-search sidebarSearchIcon"></i>
            </Nav.Link>
            <hr></hr>
            <Nav.Link>
              <SearchInput
                categoryLabel={props.categoryLabel}
                searchTerm={props.searchTerm}
                searchIsLoaded={props.searchIsLoaded}
                handleSearchChange={props.handleSearchChange}
                handleCategoryChange={props.handleCategoryChange}
                handleSearch={props.handleSearch}
                handleClearSearch={props.handleClearSearch}
              />
            </Nav.Link>
            <hr></hr>
            <Nav.Link as={Link} to="/favorites">
              <i class="fas fa-2x fa-star sidebarStarIcon"></i>
            </Nav.Link>
          </div>
        </Container>
      </Nav>
    </Col>
  );
};

export default Sidebar;
