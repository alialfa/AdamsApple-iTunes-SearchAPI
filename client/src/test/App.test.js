/**
 * Ali_Mongi-CAPSTONE_SEARCH_API-L2T21
 * @App.test.js - tests for frontend components
 */
import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import renderer from "react-test-renderer";
import Header from "../components/Header";
import Sidebar from "../components/Sidebar";
import SearchWelcome from "../components/SearchWelcome";
import SearchResults from "../components/SearchResults";
import resultsSample from "./searchresult.json";

/* test for component 1 */
it("TEST: `HEADER` COMPONENT >>> generates navigation menu as intended (1)", () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* test for component 2 */
it("TEST: `INPUT` COMPONENT >>> it generates a sidebar with an input box + option menu + cmd buttons + welcome area (2)", () => {
  const tree = renderer
    .create(
      <Router>
        <Sidebar categoryLabel="CATEGORY" />
        <SearchWelcome />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* test for component 3 */
it("TEST: `SEARCH RESULTS` COMPONENT >>> it auto-generates fictional search results from json array (3)", () => {
  let media = resultsSample.results;
  let resultsHistory = [];
  const tree = renderer
    .create(
      <Router>
        <SearchResults
          searchTerm="Mariah Carey"
          kindSelected="all"
          message="successful"
          media={media}
          resultsHistory={resultsHistory}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
