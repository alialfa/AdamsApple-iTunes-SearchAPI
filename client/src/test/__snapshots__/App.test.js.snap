// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST: \`HEADER\` COMPONENT >>> generates navigation menu as intended (1) 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-success"
      id="headerNavbar"
    >
      <a
        className="navbar-brand"
        href="/home"
        onClick={[Function]}
      >
        <h1
          style={
            Object {
              "color": "white",
            }
          }
        >
          Â 
          ADAMS
           
          <i
            class="fab fa-apple"
          />
        </h1>
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        />
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rb-event-key={null}
              disabled={false}
              href="/home"
              onClick={[Function]}
            >
              <i
                class="fas fa-2x fa-home"
              />
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              disabled={false}
              href="/search"
              onClick={[Function]}
            >
              <i
                class="fas fa-2x fa-search"
              />
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              disabled={false}
              href="/favorites"
              onClick={[Function]}
            >
              <i
                class="fas fa-2x fa-star"
              />
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>
</div>
`;

exports[`TEST: \`INPUT\` COMPONENT >>> it generates a sidebar with an input box + option menu + cmd buttons + welcome area (2) 1`] = `
Array [
  <div
    className="col-3"
  >
    <div
      className="flex-column col-md-12 d-none d-md-block nav"
      onKeyDown={[Function]}
    >
      <div
        className="container-fluid"
      >
        <div
          class="sidebar"
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/search"
            onClick={[Function]}
          >
            <i
              class="fas fa-2x fa-search sidebarSearchIcon"
            />
          </a>
          <hr />
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <div
              className="searchInput"
            >
              <input
                autoComplete="on"
                class="form-control"
                id="inputSearchBox"
                onKeyDown={[Function]}
                placeholder="Search..."
                type="text"
              />
              <div
                className="dropdown"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-primary"
                  disabled={false}
                  id="inputDropDownCategory"
                  onClick={[Function]}
                  type="button"
                >
                  CATEGORY
                </button>
              </div>
              <div
                className="inputButtons"
              >
                <button
                  className="buttonSearch btn btn-success"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  SEARCH
                </button>
                <a
                  className="buttonSearch btn btn-secondary"
                  disabled={false}
                  href="/search"
                  onClick={[Function]}
                >
                  CLEAR
                </a>
              </div>
            </div>
          </a>
          <hr />
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/favorites"
            onClick={[Function]}
          >
            <i
              class="fas fa-2x fa-star sidebarStarIcon"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="col-9"
  >
    <div
      className="searchWelcome"
    >
      <div
        className="fade alert alert-dark show"
        role="alert"
      >
        <h1>
          [ New Search Awaits ]
        </h1>
        <ul>
          <li>
            Use the side-bar on your left to search.
          </li>
          <li>
            Filter your search to a specific category.
          </li>
          <li>
            Favorite items by clicking on the star icons.
          </li>
          <li>
            Access favorites from the big star on the sidebar.
          </li>
          <li>
            Thanks for visiting 
            <i>
              Apple Adam's API 
            </i>
             :)
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;

exports[`TEST: \`SEARCH RESULTS\` COMPONENT >>> it auto-generates fictional search results from json array (3) 1`] = `
<div
  className="col-9"
>
  <div
    className="container"
  >
    <div
      className="fade resultAlert alert alert-dark show"
      role="alert"
    >
      search_for: [Mariah Carey] ||| status: [delivered!]
    </div>
    
    <div
      className="searchResults row"
    >
       
      <div
        className="col-3"
      >
        <div
          className="resultsCard card"
        >
          <img
            className="card-img-top"
            src="https://is1-ssl.mzstatic.com/image/thumb/Music4/v4/f7/04/0c/f7040cc4-be4a-8935-094e-a3e9c23eb6b2/source/1000x1000bb.jpg"
          />
          <div
            className="card-body"
          >
            <div
              class="resultsCardTitle"
              className="card-title h5"
            >
              <i>
                <h5>
                  Mariah Carey
                </h5>
              </i>
            </div>
          </div>
          <div
            className="list-group-flush list-group"
            onKeyDown={[Function]}
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Track: 
                </h6>
                <h5>
                  All I Want For Christmas Is You
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Album: 
                </h6>
                <h5>
                  Merry Christmas
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
                style={
                  Object {
                    "color": "darkred",
                  }
                }
              >
                <h6>
                  
                </h6>
                <h5>
                  $
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Genre: 
                </h6>
                <h5>
                  Holiday
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
                style={
                  Object {
                    "color": "darkred",
                  }
                }
              >
                $
                9.99
              </div>
            </div>
          </div>
        </div>
        <i
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          class="fas fa-star fa-1x resultsCardStarIcon"
          id={0}
          to="/search"
        />
      </div>
      <div
        className="col-3"
      >
        <div
          className="resultsCard card"
        >
          <img
            className="card-img-top"
            src="https://is4-ssl.mzstatic.com/image/thumb/Music1/v4/77/a5/9c/77a59c36-0134-c770-a157-9f2ab1c775cf/source/1000x1000bb.jpg"
          />
          <div
            className="card-body"
          >
            <div
              class="resultsCardTitle"
              className="card-title h5"
            >
              <i>
                <h5>
                  Mariah Carey
                </h5>
              </i>
            </div>
          </div>
          <div
            className="list-group-flush list-group"
            onKeyDown={[Function]}
          >
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Track: 
                </h6>
                <h5>
                  Fantasy
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Album: 
                </h6>
                <h5>
                  Greatest Hits
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  
                </h6>
                <h5 />
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="resultsCardItem"
                style={
                  Object {
                    "color": "darkred",
                  }
                }
              >
                <h6>
                  
                </h6>
                <h5>
                  $
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
              >
                <h6>
                  Genre: 
                </h6>
                <h5>
                  R&B/Soul
                </h5>
              </div>
            </div>
            <div
              className="list-group-item"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                className="resultsCardItem"
                style={
                  Object {
                    "color": "darkred",
                  }
                }
              >
                $
                14.99
              </div>
            </div>
          </div>
        </div>
        <i
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          class="fas fa-star fa-1x resultsCardStarIcon"
          id={1}
          to="/search"
        />
      </div>
    </div>
  </div>
</div>
`;
